### ISSUE TITLE
#road_label shield 

### ISSUE THREAD

**Issue by clkao** [osm2vectortiles/osm2vectortiles#356]
Currently `shield` seems [always 'default'](https://github.com/osm2vectortiles/osm2vectortiles/blob/master/osm2vectortiles.tm2source/data.yml#L885), where mapbox has a detailed list of [shield](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#road_label) determined by some rules.

Perhaps we can crowdsource a list of rules with `(country, roadclass, ref) -> shield-name`

**Comment by lukasmartinelli** [comment]
> Currently shield seems always 'default', where mapbox has a detailed list of shield determined by some rules.

Yes we thought about that. @manuelroth has invested some time and decided that it was too resource intensive with too little gain for us to do it.

However we are of course open to a PR.

The rules are actually quite obscure here from what I remember. But if you find a way to map these three attributes to shield-name please share. The only information we are lacking in the DB is the country information now (but we can pull this in from NaturalEarth if required).

**Comment by clkao** [comment]
I just noticed that mapbox provides [undocumented iso_3166_2](https://github.com/mapbox/vector-tiles/issues/17) which can be used in cartocss to match and provide custom shield.

So instead of a cumbersome plpgsql function to derive the sheild-name, we can start with providing the iso_3166_2 and iso_3166_1 field, and additionally the `network` field from the highway relation.

**Comment by lukasmartinelli** [comment]
Due to the licensing issues we'll not support the shield names. The shield thing is also something that is in Mapbox Streets style (not in necessarily in OSM Bright).

However adding OSM iso field / or just the country the road is in to the vector schema is something i will consider for v3.

However there is usually no iso value on the roads themselves directly right?
Then we must do a query to check in which country the road is in and use the iso value of the country.

**Comment by clkao** [comment]
@lukasmartinelli did mapbox say something about the licensing issues?  I thought they are rather liberal on their works.

re iso value: yes, it's to be derived from country or admin boundary that the roads are within.
